client/src/components/add-job-lead-component/index.jsx

```diff
@@ -0,0 +1,284 @@
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+} from "@mui/material";
+import {
+  Container,
+  ButtonContainer,
+  JobLeadContainer,
+  H3,
+  H1,
+  Body,
+  ButtonL,
+} from "./index.styles";
+
+function AddJobLead({ jobLeadData, setJobLeadData, resetInitialState }) {
+  AddJobLead.propTypes = {
+    // eslint-disable-next-line react/forbid-prop-types
+    jobLeadData: PropTypes.array.isRequired,
+    setJobLeadData: PropTypes.func.isRequired,
+    resetInitialState: PropTypes.func.isRequired,
+  };
+  const [open, setOpen] = useState(false);
+
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const handleAddJobLead = () => {
+    const newId = jobLeadData.length;
+    setJobLeadData([
+      ...jobLeadData,
+      {
+        id: newId,
+        employer: "",
+        title: "",
+        minCompensation: "",
+        maxCompensation: "",
+        hoursPerWeek: "",
+        nationalOC: "",
+        creationDate: null,
+        expirationDate: null,
+        employmentType: "",
+      },
+    ]);
+  };
+
+  const handleInputChange = (input, id, field) => {
+    const newJobLeads = jobLeadData.map((lead) =>
+      lead.id === id ? { ...lead, [field]: input } : lead,
+    );
+    setJobLeadData(newJobLeads);
+  };
+
+  const handleResetInputs = () => {
+    resetInitialState();
+  };
+
+  return (
+    <Container>
+      <H1>Adding a Job Lead</H1>
+      <Body>Input information about the job lead.</Body>
+      {jobLeadData.map((lead) => (
+        <JobLeadContainer key={lead.id}>
+          <H3>Job Lead</H3>
+          <FormControl fullWidth sx={{ m: 1, width: "96%" }}>
+            <InputLabel id="nameLabel">Employer Name</InputLabel>
+            <Select
+              sx={{ textAlign: "left" }}
+              labelId="nameLabel"
+              id={`employer${lead.id}`}
+              value={lead.employer}
+              label="Employer Name"
+              helpertext="*Required"
+              onChange={(e) =>
+                handleInputChange(e.target.value, lead.id, "employer")
+              }
+            >
+              <MenuItem value="name 1">Name 1</MenuItem>
+              <MenuItem value="name 2">Name 2</MenuItem>
+              <MenuItem value="name 3">Name 3</MenuItem>
+            </Select>
+            <FormHelperText>*Required</FormHelperText>
+          </FormControl>
+          <TextField
+            fullWidth
+            sx={{ m: 1, width: "96%" }}
+            id={`title${lead.id}`}
+            value={lead.title}
+            onChange={(e) =>
+              handleInputChange(e.target.value, lead.id, "title")
+            }
+            label="Job Title"
+            helpertext="*Required"
+          />
+          <FormControl fullWidth sx={{ m: 1, width: "47%" }}>
+            <InputLabel id="minCompensationLabel">
+              Compensation Minimum
+            </InputLabel>
+            <OutlinedInput
```

Comment:
Can we add a mask so only number and decimals can be entered

Reply:
I decided to set the type attributes of the inputs to number instead of using imask to match the add employer job lead page inputs if that's okay? It should also only accept numbers and decimals.